packages/tao/src/commands/migrate.ts

```diff
@@ -439,7 +447,23 @@ function createFetcher() {
     packageVersion: string
   ): Promise<MigrationsJson> {
     if (!cache[`${packageName}-${packageVersion}`]) {
-      const dir = dirSync().name;
+      // Setting up a tmpDir. In order to obey various .npmrc files
+      // this is a bit more complex than it should be. `.npmrc` can be present
+      // in ~/, workspace root, global, or in the npm installation directory.
+      const tmpdir = `${appRootPath}/node_modules/.tmp`;
```

Comment:
We should think about this more.

Before, this was under `/tmp` which the OS cleans up every so often. 

I think if we do this inside the workspace, we should use our cache directory... which gets cleaned up every so often.

There is a cleanup below but I think sometimes it might fail due to permissions though that might not be the case anymore :thinking: 

The directory is also not cleaned up in case of errors intentionally so it's possible to debug errors.

Reply:
Definitely open to suggestions on this part. We could move this under our cache, that shouldn't cause any issues. I've not encountered that cleanup failing, but it is correct that it would not be cleaned on an error.

Reply:
Discussed w/ Jason. Since npm doesn't crawl up file tree to resolve npmrcs, we don't need to relocate tmpDir. Npm will **_ALWAYS_** look at ~/.npmrc, regardless of where process.cwd is.